### ISSUE TITLE
java backend cannot allow cast a Set to a Term

### ISSUE THREAD

**Issue by liyili2** [kframework/k-legacy#640]
Thanks for Dwight, I make a movement in kruning all programs. However, the krun commend still cannot run a krun with parser set to binary. Here is the example:

When I run this command

```
krun --output="binary" --output-file="tmp-kcc-accum-OSK8.o" -cOPTIONS="(SetItem(NOLINK) .Set)" -cPGM=".K" --config-var-parser="kast --parser binary" -cOBJ1="tmp-kcc-accum-OSK8.o" -cOBJ2="math.o" -d c11-translation-kompiled > tmp-kcc-merge-stdout-2aOR.o 2>&1
```

K will give me some error like:

```
Exception in thread "main" java.lang.ClassCastException: org.kframework.kil.Set cannot be cast to org.kframework.backend.java.kil.Term
    at org.kframework.backend.java.symbolic.KILtoBackendJavaKILTransformer.visit(KILtoBackendJavaKILTransformer.java:279)
    at org.kframework.backend.java.symbolic.KILtoBackendJavaKILTransformer.visit(KILtoBackendJavaKILTransformer.java:78)
    at org.kframework.kil.Cell.accept(Cell.java:287)
    at org.kframework.kil.AbstractVisitor.visitNode(AbstractVisitor.java:92)
    at org.kframework.backend.java.symbolic.KILtoBackendJavaKILTransformer.visit(KILtoBackendJavaKILTransformer.java:326)
    at org.kframework.backend.java.symbolic.KILtoBackendJavaKILTransformer.visit(KILtoBackendJavaKILTransformer.java:78)
    at org.kframework.kil.Bag.accept(Bag.java:69)
    at org.kframework.kil.AbstractVisitor.visitNode(AbstractVisitor.java:92)
    at org.kframework.backend.java.symbolic.KILtoBackendJavaKILTransformer.visit(KILtoBackendJavaKILTransformer.java:269)
    at org.kframework.backend.java.symbolic.KILtoBackendJavaKILTransformer.visit(KILtoBackendJavaKILTransformer.java:78)
    at org.kframework.kil.Cell.accept(Cell.java:287)
    at org.kframework.kil.AbstractVisitor.visitNode(AbstractVisitor.java:92)
    at org.kframework.backend.java.symbolic.KILtoBackendJavaKILTransformer.visit(KILtoBackendJavaKILTransformer.java:326)
    at org.kframework.backend.java.symbolic.KILtoBackendJavaKILTransformer.visit(KILtoBackendJavaKILTransformer.java:78)
    at org.kframework.kil.Bag.accept(Bag.java:69)
    at org.kframework.kil.AbstractVisitor.visitNode(AbstractVisitor.java:92)
    at org.kframework.backend.java.symbolic.KILtoBackendJavaKILTransformer.visit(KILtoBackendJavaKILTransformer.java:269)
    at org.kframework.backend.java.symbolic.KILtoBackendJavaKILTransformer.visit(KILtoBackendJavaKILTransformer.java:78)
    at org.kframework.kil.Cell.accept(Cell.java:287)
    at org.kframework.kil.AbstractVisitor.visitNode(AbstractVisitor.java:92)
    at org.kframework.backend.java.symbolic.KILtoBackendJavaKILTransformer.visit(KILtoBackendJavaKILTransformer.java:185)
    at org.kframework.backend.java.symbolic.KILtoBackendJavaKILTransformer.visit(KILtoBackendJavaKILTransformer.java:78)
    at org.kframework.kil.KLabelInjection.accept(KLabelInjection.java:68)
    at org.kframework.kil.AbstractVisitor.visitNode(AbstractVisitor.java:92)
    at org.kframework.backend.java.symbolic.KILtoBackendJavaKILTransformer.visit(KILtoBackendJavaKILTransformer.java:165)
    at org.kframework.backend.java.symbolic.KILtoBackendJavaKILTransformer.visit(KILtoBackendJavaKILTransformer.java:78)
    at org.kframework.kil.KApp.accept(KApp.java:180)
    at org.kframework.kil.AbstractVisitor.visitNode(AbstractVisitor.java:92)
    at org.kframework.backend.java.symbolic.KILtoBackendJavaKILTransformer.visit(KILtoBackendJavaKILTransformer.java:260)
    at org.kframework.backend.java.symbolic.KILtoBackendJavaKILTransformer.visit(KILtoBackendJavaKILTransformer.java:78)
    at org.kframework.kil.KList.accept(KList.java:52)
    at org.kframework.kil.AbstractVisitor.visitNode(AbstractVisitor.java:92)
    at org.kframework.backend.java.symbolic.KILtoBackendJavaKILTransformer.visit(KILtoBackendJavaKILTransformer.java:166)
    at org.kframework.backend.java.symbolic.KILtoBackendJavaKILTransformer.visit(KILtoBackendJavaKILTransformer.java:78)
    at org.kframework.kil.KApp.accept(KApp.java:180)
    at org.kframework.kil.AbstractVisitor.visitNode(AbstractVisitor.java:92)
    at org.kframework.backend.java.symbolic.KILtoBackendJavaKILTransformer.visit(KILtoBackendJavaKILTransformer.java:175)
    at org.kframework.backend.java.symbolic.KILtoBackendJavaKILTransformer.visit(KILtoBackendJavaKILTransformer.java:78)
    at org.kframework.kil.KItemProjection.accept(KItemProjection.java:70)
    at org.kframework.kil.AbstractVisitor.visitNode(AbstractVisitor.java:92)
    at org.kframework.backend.java.symbolic.KILtoBackendJavaKILTransformer.visit(KILtoBackendJavaKILTransformer.java:185)
    at org.kframework.backend.java.symbolic.KILtoBackendJavaKILTransformer.visit(KILtoBackendJavaKILTransformer.java:78)
    at org.kframework.kil.KLabelInjection.accept(KLabelInjection.java:68)
    at org.kframework.kil.AbstractVisitor.visitNode(AbstractVisitor.java:92)
    at org.kframework.backend.java.symbolic.KILtoBackendJavaKILTransformer.visit(KILtoBackendJavaKILTransformer.java:165)
    at org.kframework.backend.java.symbolic.KILtoBackendJavaKILTransformer.visit(KILtoBackendJavaKILTransformer.java:78)
    at org.kframework.kil.KApp.accept(KApp.java:180)
    at org.kframework.kil.AbstractVisitor.visitNode(AbstractVisitor.java:92)
    at org.kframework.backend.java.symbolic.KILtoBackendJavaKILTransformer.visit(KILtoBackendJavaKILTransformer.java:260)
    at org.kframework.backend.java.symbolic.KILtoBackendJavaKILTransformer.visit(KILtoBackendJavaKILTransformer.java:78)
    at org.kframework.kil.KList.accept(KList.java:52)
    at org.kframework.kil.AbstractVisitor.visitNode(AbstractVisitor.java:92)
    at org.kframework.backend.java.symbolic.KILtoBackendJavaKILTransformer.visit(KILtoBackendJavaKILTransformer.java:166)
    at org.kframework.backend.java.symbolic.KILtoBackendJavaKILTransformer.visit(KILtoBackendJavaKILTransformer.java:78)
    at org.kframework.kil.KApp.accept(KApp.java:180)
    at org.kframework.kil.AbstractVisitor.visitNode(AbstractVisitor.java:92)
    at org.kframework.backend.java.symbolic.KILtoBackendJavaKILTransformer.visit(KILtoBackendJavaKILTransformer.java:240)
    at org.kframework.backend.java.symbolic.KILtoBackendJavaKILTransformer.visit(KILtoBackendJavaKILTransformer.java:78)
    at org.kframework.kil.KSequence.accept(KSequence.java:46)
    at org.kframework.kil.AbstractVisitor.visitNode(AbstractVisitor.java:92)
    at org.kframework.backend.java.symbolic.KILtoBackendJavaKILTransformer.visit(KILtoBackendJavaKILTransformer.java:279)
    at org.kframework.backend.java.symbolic.KILtoBackendJavaKILTransformer.visit(KILtoBackendJavaKILTransformer.java:78)
    at org.kframework.kil.Cell.accept(Cell.java:287)
    at org.kframework.kil.AbstractVisitor.visitNode(AbstractVisitor.java:92)
    at org.kframework.backend.java.symbolic.KILtoBackendJavaKILTransformer.visit(KILtoBackendJavaKILTransformer.java:326)
    at org.kframework.backend.java.symbolic.KILtoBackendJavaKILTransformer.visit(KILtoBackendJavaKILTransformer.java:78)
    at org.kframework.kil.Bag.accept(Bag.java:69)
    at org.kframework.kil.AbstractVisitor.visitNode(AbstractVisitor.java:92)
    at org.kframework.backend.java.symbolic.KILtoBackendJavaKILTransformer.visit(KILtoBackendJavaKILTransformer.java:269)
    at org.kframework.backend.java.symbolic.KILtoBackendJavaKILTransformer.visit(KILtoBackendJavaKILTransformer.java:78)
    at org.kframework.kil.Cell.accept(Cell.java:287)
    at org.kframework.kil.AbstractVisitor.visitNode(AbstractVisitor.java:92)
    at org.kframework.backend.java.symbolic.KILtoBackendJavaKILTransformer.visit(KILtoBackendJavaKILTransformer.java:326)
    at org.kframework.backend.java.symbolic.KILtoBackendJavaKILTransformer.visit(KILtoBackendJavaKILTransformer.java:78)
    at org.kframework.kil.Bag.accept(Bag.java:69)
    at org.kframework.kil.AbstractVisitor.visitNode(AbstractVisitor.java:92)
    at org.kframework.backend.java.symbolic.KILtoBackendJavaKILTransformer.visit(KILtoBackendJavaKILTransformer.java:269)
    at org.kframework.backend.java.symbolic.KILtoBackendJavaKILTransformer.visit(KILtoBackendJavaKILTransformer.java:78)
    at org.kframework.kil.Cell.accept(Cell.java:287)
    at org.kframework.kil.AbstractVisitor.visitNode(AbstractVisitor.java:92)
    at org.kframework.backend.java.symbolic.KILtoBackendJavaKILTransformer.visit(KILtoBackendJavaKILTransformer.java:326)
    at org.kframework.backend.java.symbolic.KILtoBackendJavaKILTransformer.visit(KILtoBackendJavaKILTransformer.java:78)
    at org.kframework.kil.Bag.accept(Bag.java:69)
    at org.kframework.kil.AbstractVisitor.visitNode(AbstractVisitor.java:92)
    at org.kframework.backend.java.symbolic.KILtoBackendJavaKILTransformer.visit(KILtoBackendJavaKILTransformer.java:269)
    at org.kframework.backend.java.symbolic.KILtoBackendJavaKILTransformer.visit(KILtoBackendJavaKILTransformer.java:78)
    at org.kframework.kil.Cell.accept(Cell.java:287)
    at org.kframework.kil.AbstractVisitor.visitNode(AbstractVisitor.java:92)
    at org.kframework.backend.java.symbolic.KILtoBackendJavaKILTransformer.transformTerm(KILtoBackendJavaKILTransformer.java:134)
    at org.kframework.backend.java.kil.Term.of(Term.java:50)
    at org.kframework.backend.java.symbolic.JavaSymbolicKRun.javaKILRun(JavaSymbolicKRun.java:96)
    at org.kframework.backend.java.symbolic.JavaSymbolicKRun.internalRun(JavaSymbolicKRun.java:81)
    at org.kframework.backend.java.symbolic.JavaSymbolicKRun.run(JavaSymbolicKRun.java:76)
    at org.kframework.krun.Main.normalExecution(Main.java:372)
    at org.kframework.krun.Main.execute_Krun(Main.java:1454)
    at org.kframework.main.Main.main(Main.java:64)
```

**Comment by andreistefanescu** [comment]
I am also getting this one for wcet in my fork... weird

**Comment by dwightguth** [comment]
This should be fixed. Please open a new issue if you run tnto something new.

**Comment by dwightguth** [comment]
Fixed #643
